pkg/commands/compute/language_toolchain.go

```diff
@@ -18,12 +542,77 @@ func getJsToolchainBinPath(bin string) (string, error) {
 	return strings.TrimSpace(string(path)), nil
 }
 
-func checkJsPackageDependencyExists(bin, name string) bool {
-	// gosec flagged this:
-	// G204 (CWE-78): Subprocess launched with variable
-	// Disabling as the variables come from trusted sources:
-	// The CLI parser enforces supported values via EnumVar.
-	/* #nosec */
-	err := exec.Command(bin, "list", "--json", "--depth", "0", name).Run()
-	return err == nil
+// execCommand opens a sub shell to execute the language build script.
+func execCommand(cmd string, args []string, out, progress io.Writer, verbose bool, timeout int, errlog fsterr.LogInterface) error {
```

Comment:
Previously this logic was duplicated four times across the code base. Now the logic is shared across all supported languages.